store/postgres/src/store.rs

```diff
@@ -434,6 +434,104 @@ impl Store {
 }
 
 impl StoreTrait for Store {
+    fn authorize_subgraph_name(&self, name: String, new_access_token: String) -> Result<(), Error> {
+        use db_schema::subgraph_names::dsl::*;
+
+        insert_into(subgraph_names)
+            .values((
+                subgraph_name.eq(&name),
+                subgraph_id.eq::<Option<SubgraphId>>(None),
+                access_token.eq(&new_access_token),
+            )).on_conflict(subgraph_name)
+            .do_update()
+            .set(access_token.eq(&new_access_token))
+            .execute(&*self.conn.lock().unwrap())
+            .map(|_| ())
+            .map_err(Error::from)
+    }
+
+    fn check_subgraph_name_access_token(
+        &self,
+        name: String,
+        untrusted_access_token: String,
+    ) -> Result<bool, Error> {
+        use db_schema::subgraph_names::dsl::*;
+
+        let real_access_token_opt = subgraph_names
+            .select(access_token)
+            .filter(subgraph_name.eq(&name))
+            .first::<Option<String>>(&*self.conn.lock().unwrap())?;
+
+        match real_access_token_opt {
+            None => {
+                debug!(
+                    self.logger,
+                    "Subgraph name {:?} has no associated access token. Access denied by default.",
```

Comment:
This should use structured logging:
```rust
debug!(
  self.logger,
  "Subgraph has no associated access token. Access denied by default.";
  "subgraph_name" => &subgraph_name
);
```

Also: you're passing in `access_token` instead of `subgraph_name` anyway. 😉